openshift/machine-config-operator
# Add cgroupv2 confiuration support in MCO

Signed-off-by: Sai Ramesh Vanka <svanka@redhat.com>



**- What I did**
Added the code to modify the cgroup configuration on all the nodes based on the user provided input of the [cgroupMode](https://github.com/openshift/api/blob/master/config/v1/types_node.go#L36) in the `nodes.config.openshift.io` custom resource.

**- How to verify it**
By default the `cgroupv1` is enabled on all the nodes of an OCP cluster. Edit the `cgroupMode` field of the `nodes.config.openshift.io` custom resource's `spec` and observe the following kernel arguments appended in the newly created machine config object.

```
# In case of cgroupMode set to "v1"
kernelArguments:
  - systemd.unified_cgroup_hierarchy=0
  - systemd.legacy_systemd_cgroup_controller=1
```
```
In case of cgroupMode set to "v2"
kernelArguments:
  - systemd.unified_cgroup_hierarchy=1
  - cgroup_no_v1="all"
  - psi=1
```
Also observe the `cgroupv2` output when the following command is run inside any node after the nodes are updated with the new configuration.
```
sh-4.4# stat -c %T -f /sys/fs/cgroup/
cgroup2fs
```
By default, the above command outputs `tmpfs`
Reference: https://kubernetes.io/docs/concepts/architecture/cgroups/

**- Description for the changelog**

Added support to enable `cgroupv2` configuration on all the nodes of an OCP cluster.
